src/cli/cli.cpp

```diff
@@ -2662,5 +2664,35 @@ void Interpreter::ProcessNetworkDiagnostic(int argc, char *argv[])
     AppendResult(error);
 }
 
+void Interpreter::s_HandleDiagnosticGetResponse(otMessage aMessage, const otMessageInfo *aMessageInfo,
+                                                void *aContext)
+{
+    static_cast<Interpreter *>(aContext)->HandleDiagnosticGetResponse(*static_cast<Message *>(aMessage),
+                                                                      *static_cast<const Ip6::MessageInfo *>(aMessageInfo));
+}
+
+void Interpreter::HandleDiagnosticGetResponse(Message &aMessage, const Ip6::MessageInfo &)
+{
+    sServer->OutputFormat("DIAG_GET.rsp: ");
+
+    uint8_t buf[16];
```

Comment:
nit: prefer to put variable declarations at top of containing scope (missed this in my initial review).